ADR: Home Assistant Container as supported installation method (#393)
home-assistant/architecture: adr/0013-home-assistant-container.md

# 0012. Installation method: Home Assistant Container

Date: 2020-06-03

## Status

Accepted

## Context

Define a supported installation method as per [ADR-0012](https://github.com/home-assistant/architecture/blob/master/adr/0012-define-supported-installation-method.md).

## Decision

> This is for running just the Home Assistant Core application on native OCI compatible containerization system. It does not provide the full Supervisor experience, and thus does not provide the Supervisor panel and add-ons.

Reviewer: > It does not provide the full Supervisor experience

This implies that it provides a partial Supervisor experience. Suggest removing "full" here

> This is a general installation method that is recommended as an alternative to the Home Assistant One installation method. Due to the shared image with the Home Assistant One installation method, almost all documentation applies to the Home Assistant Container as well.

Reviewer: > alternative to the Home Assistant One installation method
> Due to the shared image with the Home Assistant One installation method

Replace One with TBC or get that ADR published first ;)

Author: Published: #395.

Marked hypervisors as tbd as that was blocking it.

The only supported way to run the container is on the host network as root with full privileges.

### Supported Containerization system and version

- Any Open Container Initiative (OCI) compatible containerization system.

### Supported boards/hardware/machines

- Machines of the following architectures: amd64, i386, armhf, aarch64, armv7

### Supported Operating Systems and versions

> - Running Home Assistant Container, is only supported on Linux.

Reviewer: > Running Home Assistant Container, is only supported on Linux

Remove the comma

> - Windows, and BSD installations (e.g., macOS and FreeBSD) are not supported.

Reviewer: > Windows, and BSD

Remove the comma

### Additional notes

There is a wide variety of containerization software available. From that perspective, Home Assistant will only actively document the use of Docker.

### Required Expertise

- **Installation**
  This requires the user to have an existing system that can run Docker containers. Installation is either done by running a command from the Docker-cli or via a user interface (Synology, Portainer)

> * **Start when the system is started:** User is responsible for configuring this.

Reviewer 2: I think it's still ambiguous what we're referring to in this bullet.

Maybe:

> The user is responsible for configuring the system to start the container when the system is started.

Reviewer 2: ```suggestion
* **Start when the system is started:**
  The user is responsible for configuring the system to start the container when the system is started.
```

> * **Start when the system is started:** up to the container manager that the user uses.

Reviewer 2: What is up to the container manager? Do we mean that the home assistant container should start when the host system is started?

> * **Run with full network access:** Default installation instructions prescribe net=host but the user can diverge. Usually happens when the user decides it’s not the Docker-way.

Reviewer 2: We need to say what "Usually happens".

> * **Access USB devices:** Responsibility of user.

Reviewer 2: We need to explain this more. What about the access to USB devices is the responsibility of the user?

> * **Access USB devices:** Responsibility of user.

Reviewer: > Responsibility of user
to
> It is up to the user to ensure that all devices are correctly passed through to the container.

> * **Maintenance Home Assistant installation**

Reviewer 2: ```suggestion
* **Maintaining the Home Assistant installation**
```

>   If using the Docker-cli user needs to manually update the run command. If using a UI the user might be notified of an upgrade and automatically update. There is no rollback in case the instance does not come online after an update.

Reviewer: > If using the Docker-cli user needs
to
If using the Docker-cli the user needs

> notified of an upgrade and automatically update
Don't we normally advise against auto-updates?
> notified of an upgrade or automatically update - automatically applying updates may result in the system not coming back online.

  - **Python upgrades:** Included in the Home Assistant container
  - **Installing Python dependencies:** Included in the Home Assistant container
  - **Updating Home Assistant:** Included in the Home Assistant container

> - **Maintenance Operating System**

Reviewer 2: > Maintenance of the Operating System

or

> Maintaining the Operating System

Reviewer 2: ```suggestion
- **Maintaining the Operating System**
```
